torch/_refs/__init__.py

```diff
@@ -1771,7 +1771,7 @@ def _reduction(
         return _safe_copy_out(copy_from=result, copy_to=out)  # type: ignore[arg-type]
 
     if result.dtype != result_dtype and result_dtype is not None:
-        result = prims.convert_element_type(result, result_dtype)
+        result = _maybe_convert_to_dtype(result, result_dtype)
```

Comment:
This conversion is known to occur because the dtypes are compared in the previous line

Reply:
Thanks for the explanation! Got that.